### DISCUSSION TITLE
NextJS app trying to read auth from getServerSideProps and returning an error

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#6731] [Questions] [answered]
Hi, I'm trying to get the logged in user from getServerSideProps (a server-side function) in a NextJS app. I've tried:

await `supabaseClient.auth.api.getUserByCookie()`

which returns an error:

`Error: No cookie found!`

On the client, though, `await supabaseClient.auth.user()` successfully returns a user.

Am I missing something?

Here is the full function:

```js
import { useEffect } from 'react';
import { Explorer } from '../components'
import { supabaseClient } from '../lib';

const Index = () => {
  
  useEffect((e) => {
    const fetchUser = async () => {
     const user2 = await supabaseClient.auth.user();
      console.log("USER CLIENT", user2)
    }

    fetchUser();
  }, [])

  return (
    <Explorer />
  );
}

export async function getServerSideProps({ req, res }) {
  const user1 = await supabaseClient.auth.api.getUserByCookie(req, res);
  const user2 = await supabaseClient.auth.user();
  console.log("USER", user1, user2)

  return {
    props: {}, // will be passed to the page component as props
  }
}

export default Index
```

**Comment by Ziinc** [comment]


**Reply by jaredcosulich** [reply]
Is it possible to get the user's JWT after the user has signed in using a magic link? Does Supabase pass the JWT in a header after authentication?

**Comment by thorwebdev** [comment, marked as answer]
For Nextjs SSR please use: https://github.com/supabase-community/auth-helpers/blob/main/packages/nextjs/README.md#server-side-rendering-ssr---withpageauth

**Reply by jaredcosulich** [reply]
Am I correct in understanding that immediately after the user clicks a magic link this will not work?

When I try `withPageAuth` on the server directly after the user clicks a magic link to sign in I am redirected to the unsigned in page (e.g. `redirectTo: '/'`)

Here is the code. After clicking a magic link I am redirected to `/auth`:

```js
export const getServerSideProps = withPageAuth({
  redirectTo: '/auth',
  async getServerSideProps(ctx) {
    return { props: { } };
  }
});
```

**Reply by thorwebdev** [reply]
Yes, correct. You will need to to back to a client page with the `UserProvider` first: https://github.com/supabase-community/auth-helpers/blob/main/packages/nextjs/README.md#basic-setup

Because the gotrue-js client needs to forward to session to the callback API route to set the cookie on the server. Once the `user` object in `useUser` is set, you can forward to SSR pages, see example here: https://github.com/vercel/nextjs-subscription-payments/blob/main/pages/signin.tsx#L58-L62

**Reply by jaredcosulich** [reply]
Ok, thank you for clarifying.

**Reply by adrianhorning08** [reply]
@jaredcosulich did you ever figure this out? I am struggling a lot with this. Seems like these supabase helper functions are not working.

**Reply by adrianhorning08** [reply]
@thorwebdev do you have some updated examples? Most of these links don't work ☹️

**Reply by thorwebdev** [reply]
I'e updated the link above, the nextjs specific content is now in a package within the monorepo: https://github.com/supabase-community/auth-helpers/blob/main/packages/nextjs

**Comment by jsonpreet** [comment]
Its not working with 

export const getServerSideProps = withPageAuth({ redirectTo: '/login' });

It show 500 error
